httpevents/request.go

```diff
@@ -75,5 +75,19 @@ func (r *request) log(logger *events.Logger, depth int) {
 	// transport outside of the httpevents package.
 	l := *logger
 	l.CallDepth += depth + 1
-	l.Log(string(fmt), arg...)
+
+	switch {
+	case is4xx(r.status) || is5xx(r.status):
+		l.Log(string(fmt), arg...)
```

Comment:
I've need to do this in a few other places. I wonder if there's a case to be made for having a convenience like: `l.track(LOG|DEBUG, ...)`.